modules/500-upmeter/images/upmeter/pkg/probe/checker/k8s_nodegroup.go

```diff
@@ -0,0 +1,232 @@
+			return check.ErrFail("expected at least %d healthy, but got %d in zone %q",
+				minReadyExpected, healthy, zone)
+		}
+	}
+
+	return nil
+}
+
+func (c *nodesByNodegroupCountChecker) listNodes() ([]v1.Node, error) {
+	timeoutSeconds := int64(c.requestTimeout.Seconds())
+	nodeList, err := c.access.Kubernetes().CoreV1().Nodes().List(metav1.ListOptions{
+		LabelSelector:  "node.deckhouse.io/group=" + c.name,
+		TimeoutSeconds: &timeoutSeconds,
+	})
+	if err != nil {
+		return nil, err
+	}
+	return nodeList.Items, nil
+}
+
+type nodeGroupFetcher struct {
+	access  kubernetes.Access
+	timeout time.Duration
+}
+
+type nodegroupProps struct {
+	minPerZone            int64
+	maxUnavailablePerZone int64
+	zones                 []string
+}
+
+func (f *nodeGroupFetcher) GetNodeGroup(name string) (nodegroupProps, error) {
+	// TODO: we fetch only v1, should we care about other versions?
+	nodegroup, err := f.access.Kubernetes().Dynamic().Resource(schema.GroupVersionResource{
+		Group:    "deckhouse.io",
+		Version:  "v1",
+		Resource: "nodegroups",
+	}).Get(name, metav1.GetOptions{})
+	if err != nil {
+		return nodegroupProps{}, err
+	}
+
+	return f.parseProps(nodegroup)
+}
+
+func (f *nodeGroupFetcher) parseProps(nodegroup *unstructured.Unstructured) (nodegroupProps, error) {
+	/*
+		  NodeGroup
+			spec:
+			  cloudInstances:
+				zones: []string
+				minPerZone: int64
+				maxPerZone: int64
+				maxUnavailablePerZone: int64
+				maxSurgePerZone: int64
+	*/
+
+	var (
+		props nodegroupProps
+		err   error
+
+		name = nodegroup.GetName()
+	)
+
+	props.minPerZone, err = f.parseNodegroupInstanceLimit(nodegroup, "minPerZone")
+	if err != nil {
+		return props, fmt.Errorf("parse minPerZone in nodegroup %q: %v", name, err)
+	}
+
+	props.maxUnavailablePerZone, err = f.parseNodegroupInstanceLimit(nodegroup, "maxUnavailablePerZone")
+	if err != nil {
+		return props, fmt.Errorf("parse maxUnavailablePerZone in nodegroup %q: %v", name, err)
+	}
+
+	props.zones, err = f.parseZones(nodegroup)
+	if err != nil {
+		return props, fmt.Errorf("parse zones in nodegroup %q: %v", name, err)
+	}
+
+	return props, nil
+}
+
+func (f *nodeGroupFetcher) parseZones(ng *unstructured.Unstructured) ([]string, error) {
+	value, ok, err := unstructured.NestedStringSlice(ng.UnstructuredContent(), "spec", "cloudInstances", "zones")
+	if err != nil {
+		return nil, err
+	}
+	if !ok {
+		//  all zones are included implicitly
+	}
+	return value, nil
+}
+
+func (f *nodeGroupFetcher) parseNodegroupInstanceLimit(ng *unstructured.Unstructured, field string) (int64, error) {
+	value, ok, err := unstructured.NestedInt64(ng.UnstructuredContent(), "spec", "cloudInstances", field)
+	if err != nil {
+		return 0, err
+	}
+	if !ok {
+		value = 0
+	}
+	return value, nil
+}
+
+func (f *nodeGroupFetcher) CountHealthyNodesByZone(nodeGroup string) (map[string]int64, error) {
+	timeoutSeconds := int64(f.timeout.Seconds())
+	nodeList, err := f.access.Kubernetes().CoreV1().Nodes().List(metav1.ListOptions{
```

Comment:
It looks like it would be best to use an informer for this purpose because upmeter lists and gets objects (operations that infromerts tend to optimize). 

Is this a complicated change? Can we switch to the informer in the scope of this PR, or at least leave a TODO message?